CMakeLists.txt

```diff
@@ -0,0 +1,242 @@
+cmake_minimum_required(VERSION 3.16)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
+
+# See release_checklist.md
+set(MAJOR_VERSION 2)
+set(MINOR_VERSION 1)
+set(MICRO_VERSION 0)
+set(SDL_REQUIRED_VERSION 2.0.0)
+
+set(DYLIB_COMPATIBILITY_VERSION "1")
+
+include(PrivateSdlFunctions)
+sdl_calculate_derived_version_variables()
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
+    message(FATAL_ERROR "Prevented in-tree built. Please create a build directory outside of the SDL_net source code and call cmake from there")
+endif()
+
+project(SDL2_net
+    LANGUAGES C
+    VERSION "${FULL_VERSION}"
+)
+
+# Set defaults preventing destination file conflicts
+set(SDL2NET_DEBUG_POSTFIX "d"
+    CACHE STRING "Name suffix for debug builds")
+mark_as_advanced(SDL2NET_DEBUG_POSTFIX)
+
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
+option(CMAKE_POSITION_INDEPENDENT_CODE "Build static libraries with -fPIC" ON)
+option(BUILD_SHARED_LIBS "Build the library as a shared library" ON)
+
+option(SDL2NET_INSTALL "Enable SDL2_net install target" ON)
+option(SDL2NET_SHOWINTERFACES "Build the showinterfaces test program" OFF)
+
+# Save BUILD_SHARED_LIBS variable
+set(SDL2NET_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+
+if(SDL2NET_BUILD_SHARED_LIBS)
+    set(sdl2_net_export_name SDL2_net)
+    set(sdl2_net_install_name_infix shared)
+    set(sdl2_target_name SDL2::SDL2)
+else()
+    set(sdl2_net_export_name SDL2_net-static)
+    set(sdl2_net_install_name_infix static)
+    set(sdl2_target_name SDL2::SDL2-static)
+endif()
+
+sdl_find_sdl2(${sdl2_target_name} ${SDL_REQUIRED_VERSION})
+
+add_library(SDL2_net
+    SDLnet.c
+    SDLnetselect.c
+    SDLnetTCP.c
+    SDLnetUDP.c
+)
+add_library(SDL2_net::${sdl2_net_export_name} ALIAS SDL2_net)
+target_include_directories(SDL2_net PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>"
+)
+target_compile_definitions(SDL2_net PRIVATE
+    BUILD_SDL
+    SDL_BUILD_MAJOR_VERSION=${MAJOR_VERSION}
+    SDL_BUILD_MINOR_VERSION=${MINOR_VERSION}
+    SDL_BUILD_MICRO_VERSION=${MICRO_VERSION}
+)
+target_link_libraries(SDL2_net PRIVATE ${sdl2_target_name})
+if(WIN32)
+    if (MSVC)
+        set_target_properties(SDL2_net PROPERTIES COMPILE_FLAGS "/wd4244 /wd4996")
```

Comment:
```suggestion
        target_compile_options(SDL2_net PRIVATE /wd4244 /wd4996)
```

Reply:
Thank you for the suggestion, I applied your suggestion in 90a143d. While also removing `/wd4996` (redundant with `_WINSOCK_DEPRECATION_NO_WARNINGS`) and adding `/W3` (to be consistent with the solution shipped in the `VisualC/` folder).